doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,291 @@ Reasoning with inductive types
+   value `i`, it uses :n:`apply c__i`, where :n:`c__i` is the i-th constructor
+   of :g:`I`.  If not specified, the tactic tries all the constructors,
+   which can result in more than one success (e.g. for `\\/`) when using compound
+   backtracking tactics such as `constructor; ...`.
+
+   :n:`{? with @bindings }`
+     If specified, the :n:`apply` is done as :n:`apply … with @bindings`.
+
+     .. warning::
+
+        The terms in :token:`bindings` are checked in the context
+        where constructor is executed and not in the context where :tacn:`apply`
+        is executed (the introductions are not taken into account).
 
    .. exn:: Not an inductive product.
       :undocumented:
 
    .. exn:: Not enough constructors.
       :undocumented:
 
-   .. tacv:: constructor
-
-      This tries :g:`constructor 1` then :g:`constructor 2`, ..., then
-      :g:`constructor n` where ``n`` is the number of constructors of the head
-      of the goal.
-
-   .. tacv:: constructor @natural with @bindings
-
-      Let ``c`` be the i-th constructor of :g:`I`, then
-      :n:`constructor i with @bindings` is equivalent to
-      :n:`intros; apply c with @bindings`.
-
-      .. warning::
-
-         The terms in :token:`bindings` are checked in the context
-         where constructor is executed and not in the context where :tacn:`apply`
-         is executed (the introductions are not taken into account).
-
-   .. tacv:: split {? with @bindings }
-      :name: split
-
-      This applies only if :g:`I` has a single constructor. It is then
-      equivalent to :n:`constructor 1 {? with @bindings }`. It is
-      typically used in the case of a conjunction :math:`A \wedge B`.
+   .. exn: The type has no constructors.
+      :undocumented:
 
-      .. tacv:: exists @bindings
-         :name: exists
+.. tacn:: split {? with @bindings }
 
-         This applies only if :g:`I` has a single constructor. It is then equivalent
-         to :n:`intros; constructor 1 with @bindings.` It is typically used in
-         the case of an existential quantification :math:`\exists x, P(x).`
+   Equivalent to :n:`constructor 1 {? with @bindings }` when the
+   conclusion is an inductive type with a single constructor.  The :n:`@bindings`
+   specify any parameters required for the constructor. It is
+   typically used to split conjunctions in the conclusion such as `A /\\ B` into
+   two new goals `A` and `B`.
 
-      .. tacv:: exists {+, @bindings }
+.. tacn:: exists {*, @bindings }
 
-         This iteratively applies :n:`exists @bindings`.
+   Equivalent to :n:`intros` followed by :n:`constructor 1 {? with binding__i }`
+   for each binding when the conclusion is an
+   inductive type with a single constructor.  It is typically used on
+   existential quantifications in the form `exists x, P x.`
 
-      .. exn:: Not an inductive goal with 1 constructor.
-         :undocumented:
+.. exn:: Not an inductive goal with 1 constructor.
+   :undocumented:
```

Comment:
I don't like having error messages like this non-indented because it's not clear to what tactic they are attached. I would prefer to see it attached to one of the two tactics and the other one refer to the error message with `:exn:`.

Reply:
I indented it.  Not sure what you have in mind for "... the other one refer to the error message with `:exn:`".

Reply:
The fact that this error message can also be triggered by `split`. Therefore, the description of `split` could contain a reference to this error message. But not very important.